server/base/src/main/java/org/apache/accumulo/server/manager/state/TabletManagementParameters.java

```diff
@@ -0,0 +1,244 @@
+      TabletManagementParameters.this.tserverGroups.forEach((resourceGroup, tservers) -> tservers
+          .forEach(tserver -> resourceGroups.put(tserver, resourceGroup)));
+      return Map.copyOf(resourceGroups);
+    });
+  }
+
+  private TabletManagementParameters(JsonData jdata) {
+    this.managerState = jdata.managerState;
+    this.parentUpgradeMap = Map.copyOf(jdata.parentUpgradeMap);
+    this.onlineTables = jdata.onlineTables.stream().map(TableId::of).collect(toUnmodifiableSet());
+    this.onlineTservers =
+        jdata.onlineTservers.stream().map(TServerInstance::new).collect(toUnmodifiableSet());
+    this.serversToShutdown =
+        jdata.serversToShutdown.stream().map(TServerInstance::new).collect(toUnmodifiableSet());
+    this.migrations = jdata.migrations.entrySet().stream()
+        .collect(toUnmodifiableMap(entry -> JsonData.strToExtent(entry.getKey()),
+            entry -> new TServerInstance(entry.getValue())));
+    this.level = jdata.level;
+    this.compactionHints = makeImmutable(jdata.compactionHints);
+    this.tserverGroups = jdata.tserverGroups.entrySet().stream().collect(toUnmodifiableMap(
+        Map.Entry::getKey,
+        entry -> entry.getValue().stream().map(TServerInstance::new).collect(toUnmodifiableSet())));
+    this.resourceGroups = Suppliers.memoize(() -> {
+      Map<TServerInstance,String> resourceGroups = new HashMap<>();
+      TabletManagementParameters.this.tserverGroups.forEach((resourceGroup, tservers) -> tservers
+          .forEach(tserver -> resourceGroups.put(tserver, resourceGroup)));
+      return Map.copyOf(resourceGroups);
+    });
+  }
+
+  public ManagerState getManagerState() {
+    return managerState;
+  }
+
+  public boolean isParentLevelUpgraded() {
+    return parentUpgradeMap.get(level);
+  }
+
+  public Set<TServerInstance> getOnlineTsevers() {
+    return onlineTservers;
+  }
+
+  public Set<TServerInstance> getServersToShutdown() {
+    return serversToShutdown;
+  }
+
+  public boolean isTableOnline(TableId tableId) {
+    return onlineTables.contains(tableId);
+  }
+
+  public Map<KeyExtent,TServerInstance> getMigrations() {
+    return migrations;
+  }
+
+  public Ample.DataLevel getLevel() {
+    return level;
+  }
+
+  public String getResourceGroup(TServerInstance tserver) {
+    return resourceGroups.get().get(tserver);
+  }
+
+  public Map<String,Set<TServerInstance>> getGroupedTservers() {
+    return tserverGroups;
+  }
+
+  public Set<TableId> getOnlineTables() {
+    return onlineTables;
+  }
+
+  public Map<Long,Map<String,String>> getCompactionHints() {
+    return compactionHints;
+  }
+
+  private static Map<Long,Map<String,String>>
+      makeImmutable(Map<Long,Map<String,String>> compactionHints) {
+    var copy = new HashMap<Long,Map<String,String>>();
+    compactionHints.forEach((ftxid, hints) -> copy.put(ftxid, Map.copyOf(hints)));
+    return Collections.unmodifiableMap(copy);
+  }
+
+  private static class JsonData {
```

Comment:
Would something less generic than JsonData help when it is declared / used?  Maybe TMParmsJson or even ParamsJason?

Reply:
IMO that's not necessary because its a private inner class and is only used inside the class and never seen outside of the class.